### ISSUE TITLE
Cityscapes training and joint training

### ISSUE THREAD

**Issue by Jumponthemoon** [yasserben/CLOUDS#4]
Hi,

I'm running the training code for training cityscapes and eval on acdc. I encountered some problems:

1.For warm up training, i found the training is not corrected conducted, it directly jump to inference step(same for gta5 training):

![image](https://github.com/yasserben/CLOUDS/assets/39290403/10ff20b4-9efe-4f54-b308-f5c7a0bdf50a)

After training, i will get a saved pth file like this:
![image](https://github.com/yasserben/CLOUDS/assets/39290403/8e61ca26-08a0-4506-a501-f4931ce20120)

I'm wondering is it normal and what's objective of warmup training?

2.For joint training, i found it requires load pretrained weight in config file:
![image](https://github.com/yasserben/CLOUDS/assets/39290403/1dc64d9d-e296-427e-b3bc-906b279fb22a)
I didn't find it in the repo. So i assume it may be the warmup checkpoint and set it accordingly. But it throw out error like this:
![image](https://github.com/yasserben/CLOUDS/assets/39290403/a7b4f5ed-3020-4706-a9c9-fc9ac3b7bdae)

Can you help me fix it? Much appreciated!

**Comment by yasserben** [comment]
Hi!

1. Before starting the training, I performed an evaluation step during both the warm-up and joint training phases. This was not essential, but I implemented it as a sanity check to ensure that the correct warm-up checkpoint is used in the joint training. Typically, during the warm-up phase, you must specify the `OUTPUT_DIR`. This directory is where all the checkpoints will be stored.
2. The parameter `MODEL.CLOUDS.WEIGHTS` should be set to point to the checkpoint file located in the `OUTPUT_DIR` you specified during the warm-up phase.

I hope this answers your question! Please feel free to ask if you have any more questions.

**Comment by Jumponthemoon** [comment]
Hi,

Thanks for your reply!

I set the MODEL.CLOUDS.WEIGHTS to be the warm-up phase trained model, but here is the issue i encountered: 
![image](https://github.com/yasserben/CLOUDS/assets/39290403/7333c069-6b8c-4136-976b-2a3d377f72e1)
I don't know what's wrong.

I'm wondering what's the role of loading warm-up checkpoint in the joint training, and can I do joint training without it if not necessary?

**Comment by yasserben** [comment]
Hi !

Can you please share the exact path of the checkpoint you set when doing joint-training ? Because normally the path should point to the model_final.pth which is stored in the output folder you det in the warm-up stage.

The warm-up stage is necessary before joint training, it's a way to get reliable pseudo labels as a starting point.

**Comment by Jumponthemoon** [comment]
Hi,

The saved checkpoint name is 'sem_seg_evaluation.pth' after warm-up training. So i set it to be:
![image](https://github.com/yasserben/CLOUDS/assets/39290403/13531f76-a840-4ea9-9647-ff6e2a6c9de0)

**Comment by yasserben** [comment]
Hi,

Sorry for the late reply; I was unavailable earlier. 😅

The path you've set for `MODEL.CLOUDS.WEIGHTS` isn't correct for joint training. I've done a warmup on Cityscapes to ensure the code is bug-free, and it works well. Here's the command you should use:

```
python train_net.py --num-gpus 2 --config-file configs/warmup/cityscapes/train_city.yaml OUTPUT_DIR ./outputs/cityscapes_warmup`
```
After the warmup, you'll find `model_final.pth` in `./outputs/cityscapes_warmup` 
![proof_github_issue](https://github.com/yasserben/CLOUDS/assets/22299630/e59c3a75-6a0c-46b1-abc9-f4b934cde8ad)

Set `MODEL.CLOUDS.WEIGHTS` to the full path of `model_final.pth` in your joint training config at `./configs/joint_training/cityscapes/train_jt_city.yaml`. Then, launch the joint training with:

```
python train_net.py --num-gpus 2 --config-file configs/joint_training/cityscapes/train_jt_city.yaml OUTPUT_DIR ./outputs/cityscapes_joint_training
```

I hope this clears things up!

Best regards.
